### PR TITLE
Trim req/resp streams by a field each

### PR DESCRIPTION
**Pull request by benaadams**
Can only be in one of the states; not both

### CODE REVIEW

File: src/Microsoft.AspNet.Server.Kestrel/Http/FrameRequestStream.cs [hunk 1]
```diff
@@ -145,14 +116,38 @@ public void StopAcceptingReads()
...
         public void Abort()
         {
             // We don't want to throw an ODE until the app func actually completes.
             // If the request is aborted, we throw an IOException instead.
-            _aborted = true;
+            if (_state != StreamState.Disposed)
+            {
+                _state = StreamState.Aborted;
+            }
+        }
```
Comment (halter73) [inline src/Microsoft.AspNet.Server.Kestrel/Http/FrameRequestStream.cs]: Nit: needs newline

File: src/Microsoft.AspNet.Server.Kestrel/Http/FrameResponseStream.cs [hunk 2]
```diff
@@ -110,14 +81,39 @@ public void StopAcceptingWrites()
...
+            switch (_state)
+            {
+                case StreamState.Open:
+                    return;
+                case StreamState.Disposed:
+                    throw new ObjectDisposedException(nameof(FrameResponseStream));
+                case StreamState.Aborted:
+                    throw new IOException("The request has been aborted.");
+                default:
+                    throw new IOException("Invalid Stream State.");
```
Comment (halter73) [inline src/Microsoft.AspNet.Server.Kestrel/Http/FrameResponseStream.cs]: Does the compilere require this considering StreamState has only 3 defined values that are already checked?

File: src/Microsoft.AspNet.Server.Kestrel/Http/FrameRequestStream.cs (hunk 1 above)
Comment (benaadams) [inline src/Microsoft.AspNet.Server.Kestrel/Http/FrameRequestStream.cs]: Done

File: src/Microsoft.AspNet.Server.Kestrel/Http/FrameResponseStream.cs (hunk 2 above)
Comment (benaadams) [inline src/Microsoft.AspNet.Server.Kestrel/Http/FrameResponseStream.cs]: Remove

**Pull request merged**